service/gcs/main.go

```diff
@@ -16,11 +16,57 @@ import (
 	"github.com/Microsoft/opengcs/service/gcs/runtime/runc"
 	"github.com/Microsoft/opengcs/service/gcs/transport"
 	"github.com/containerd/cgroups"
+	cgroupstats "github.com/containerd/cgroups/stats/v1"
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
 )
 
+func memoryLogFormat(metrics *cgroupstats.Metrics) logrus.Fields {
+	return logrus.Fields{
+		"memoryUsage":      metrics.Memory.Usage.Usage,
+		"memoryUsageMax":   metrics.Memory.Usage.Max,
+		"memoryUsageLimit": metrics.Memory.Usage.Limit,
+		"swapUsage":        metrics.Memory.Swap.Usage,
+		"swapUsageMax":     metrics.Memory.Swap.Max,
+		"swapUsageLimit":   metrics.Memory.Swap.Limit,
+		"kernelUsage":      metrics.Memory.Kernel.Usage,
+		"kernelUsageMax":   metrics.Memory.Kernel.Max,
+		"kernelUsageLimit": metrics.Memory.Kernel.Limit,
+	}
+}
+
+func readMemoryEvents(efdFile *os.File, cgName string, threshold int64, cg cgroups.Cgroup) {
+	// Buffer must be >= 8 bytes for eventfd reads
+	// http://man7.org/linux/man-pages/man2/eventfd.2.html
+	buf := make([]byte, 8)
+	for {
+		_, err := efdFile.Read(buf)
+		if err != nil {
+			logrus.WithError(err).Warn("failed to read from eventfd")
```

Comment:
I would probably make this error level, since it's a clear indication that something has failed.

Reply:
Also, we should log `cgName` here.